plugins/sed.py

```diff
@@ -0,0 +1,50 @@
+#
+# plugins/interject.py
+# Copyright (C) 2017 Josh Wolfe
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+from xyzzy.plugin import Plugin
+import re
+
+class SedPlugin(Plugin):
+    PLUGIN_COMMAND = 'sed'
+
+    def run(self, arguments):
+        flag = False
+        for event in reversed(arguments.room.get_events()):
+            if event["sender"] == arguments.current_event["sender"]:
+                if flag:
+                    return {
+                        'type': 'markdown',
+                        'content': '**CORRECTION**: %s' % self.parse_replacement(arguments, event)
+                    }
+                else:
+                    flag = True
+
+        return {
+            'type': 'text',
+            'content': ''
+        }
+
+    def accept(self, arguments):
+        if arguments.raw[0] == 's':
```

Comment:
I think it would be better to have a regex that detects a sed-like expression instead of checking the first character of the message.